AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1793,7 +1793,8 @@ - (BOOL)placeholderShouldPersist
 
 - (BOOL)_shouldHavePlaceholderLayer
 {
-  return (_placeholderEnabled && [self __implementsDisplay]);
+  CGSize layoutSize = (_layout ? _layout.size : CGSizeZero);
```

Comment:
Does this function need a lock?

 I believe it is not sufficient to check the layout size here. An element that is positioned with manual layout may have a zero size here, but a very non-zero bounds size. 

  Is this method always called on the main thread, presumably yes because the placeholder layer must be created there? If so, we should probably assert that as well, and we may be able to rely on the fact that a layout sublayers pass has occurred… Which would mean that it is sufficient to on the check bounds  and not worry about checking the layout as well